### PR TITLE
created dto, mapper and controller for books

### PR DESCRIPTION
**Pull request by SalnikovMarkiian**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/books")
```
Comment (RomanYuzviak) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]: it`s better use api/books I think

**Review by RomanYuzviak — APPROVED** [review]
Good solution!

File: pom.xml [hunk 2]
```diff
@@ -72,10 +80,40 @@
...
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>1.18.30</version>
+                        </path>
+                        <path>
+                            <groupId>org.mapstruct</groupId>
+                            <artifactId>mapstruct-processor</artifactId>
+                            <version>1.5.5.Final</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: You can use version from your properties ${org.mapstruct.version}

File: src/main/java/com/example/onlinebookstore/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/mapper/BookMapper.java]: toBook will be better

File: src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java [hunk 4]
```diff
@@ -9,15 +13,28 @@
...
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> getAll() {
-        return bookRepository.findAll();
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.getById(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.getById(id)
         .orElseThrow(() -> new EntityNotFoundException("Cannot find book by id " + id));
```

File: src/main/java/com/example/onlinebookstore/config/MapperConfig.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/config/MapperConfig.java:11]: You can use mapper config directly in BookMapper with @Mapper annotation

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/com/example/onlinebookstore/repository/BookRepository.java [hunk 6]
```diff
@@ -2,9 +2,12 @@
...
 import com.example.onlinebookstore.model.Book;
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Optional<Book> getById(Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/repository/BookRepository.java]:
```suggestion
    Optional<Book> findById(Long id);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Comments not fixed

File: src/main/java/com/example/onlinebookstore/config/MapperConfig.java (hunk 5 above)
Comment (SalnikovMarkiian) [reply src/main/java/com/example/onlinebookstore/config/MapperConfig.java:11]: I didn't quite understand what you wrote about. What exactly to take and where to insert @Mapper

Comment (Elena-Bruyako) [reply src/main/java/com/example/onlinebookstore/config/MapperConfig.java:11]: No problem) 
I meant, that config 
        componentModel = "spring",
        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
        implementationPackage = "<PACKAGE_NAME>.impl"
        
could be used directly in BookMapper class in Mapper annotation. 

It's optional, you can leave as it is.

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**